src/data/onChainPoolDataViaReadContract.ts

```diff
@@ -18,25 +37,6 @@ type Result =
 
 type Results = Result[];
 
-const abi = parseAbi([
-    'function getPoolTokens(bytes32 poolId) view returns (address[] tokens, uint256[] balances, uint256 lastChangeBlock)',
-    'function getSwapFeePercentage() view returns (uint256)',
-    'function percentFee() view returns (uint256)',
-    'function protocolPercentFee() view returns (uint256)',
-    'function getNormalizedWeights() view returns (uint256[])',
-    'function totalSupply() view returns (uint256)',
-    'function getVirtualSupply() view returns (uint256)',
-    'function getActualSupply() view returns (uint256)',
-    'function getTargets() view returns (uint256 lowerTarget, uint256 upperTarget)',
-    'function getTokenRates() view returns (uint256, uint256)',
-    'function getWrappedTokenRate() view returns (uint256)',
-    'function getAmplificationParameter() view returns (uint256 value, bool isUpdating, uint256 precision)',
-    'function getPausedState() view returns (bool)',
-    'function inRecoveryMode() view returns (bool)',
-    'function getRate() view returns (uint256)',
-    'function getScalingFactors() view returns (uint256[] memory)',
-]);
-
 // Extract the functionName property values into a union type
 type FunctionNameUnion = typeof abi[number]['name'];
```

Comment:
I checked out your branch and it seems that this type is not being computed properly.
Can you fix it?
If it starts taking too long, I'd say it's ok to leave as is, since all tests and lint checks are passing.